ambari-server/src/main/java/org/apache/ambari/server/checks/CheckDescription.java

```diff
@@ -212,4 +244,42 @@ public String getText() {
   public String getFail(String key) {
     return m_fails.containsKey(key) ? m_fails.get(key) : "";
   }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public int hashCode() {
+    return Objects.hash(m_name);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public boolean equals(Object object) {
+    if (null == object) {
+      return false;
+    }
+
+    if (this == object) {
+      return true;
+    }
+
+    if (object.getClass() != getClass()) {
+      return false;
+    }
+
+    CheckDescription that = (CheckDescription) object;
+
+    return Objects.equals(m_name, that.m_name);
```

Comment:
should not hashcode / equals / toString be based also on m_description and m_fails values?

Reply:
No, since we're basically saying the name is what makes them unique.